### PR TITLE
Docker Compose Conversion

### PR DESCRIPTION
**Pull request by Jbsco**
This pull request should follow #24. These changes convert the current Docker scripts into a unified Docker Compose script which manages Docker container operations with arguments such as `start`, `stop`, `login`, `push`, `build`, and `remove`. The Docker configuration is contained within the `compose.yaml` file and is provided as an argument to the `docker-compose` command contained within the unified script.

### CODE REVIEW

**Comment by Jbsco** [comment]
This is functional on Linux, but testing on Windows using `adamant_env.sh start` results in a container with empty `/adamant` and `/adamant_example` directories after logging in. Some projects (such as COSMOS) provide an alternative `.bat` script which parallels the shell script. Perhaps `$OSTYPE` casing is needed to allow this to work on non-Linux systems.

**Comment by Jbsco** [comment]
> [...] using `adamant_env.sh start` results in a container with empty `/adamant` and `/adamant_example` directories after logging in. [...]

This is an issue with pathing in the `compose.yaml` configuration and was fixed in [1e92cc4](https://github.com/lasp/adamant_example/pull/25/commits/1e92cc4f69720963609ec38ad746a9d9613fcb5c).

File: docker/compose.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+              source: ../../adamant
+              target: /home/user/adamant
+            - type: bind
+              source: ../../adamant_example
+              target: /home/user/adamant_example
+        extra_hosts:
+            - host.docker.internal:host-gateway
+        network_mode: host
+        image: ${DOCKER_IMAGE_NAME}
+        command: sleep infinity
```
Comment (dinkelk) [inline docker/compose.yml]: I think the command should actually run the `activate.sh` script to pre-install a lot of the dependencies for the user prior to login. however we only want this command to be run when we create the container... not when we start the container. I am not sure if this is what `command` does or not.

Comment (Jbsco) [reply docker/compose.yml]: I believe `command` in compose.yml can handle `activate.sh` - I'll test and push that change if this is the case.

File: docker/compose.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: adamant_example
+services:
+    adamant_example:
+        container_name: ${DOCKER_CONTAINER_NAME}
+        volumes:
+            - type: bind
+              source: ../../adamant
```
Comment (dinkelk) [inline docker/compose.yml]: while this is a decent assumption, we should allow users to specify where adamant lives if it is elsewhere. So I am thinking best to pull this path from `docker_config.sh`

File: docker/compose.yml (hunk 1 above)
Comment (dinkelk) [reply docker/compose.yml]: After looking around, we may need a bit more logic to only do this only the first time the container is run.

```
    command: >
      bash -c "if [ ! -f /initialized ]; then
                  # Run your one-time initialization command here
                  touch /initialized;
              fi;
```

but maybe you will find something better

**Review by dinkelk — CHANGES_REQUESTED** [review]

File: docker/compose.yml (hunk 1 above)
Comment (Jbsco) [reply docker/compose.yml]: This works when following `docker compose up -d` with `docker compose exec` to run activation. Addressed in [2d78d6d](https://github.com/lasp/adamant_example/pull/25/commits/2d78d6d0c8e1d19c370184ae2d718b1cc1bfff5b).

File: docker/compose.yml (hunk 2 above)
Comment (Jbsco) [reply docker/compose.yml]: Addressed in [802c98c](https://github.com/lasp/adamant_example/pull/25/commits/802c98cf63599489f2388c74898305c49c3f276d).

**Review by dinkelk — APPROVED** [review]

**Pull request merged**